impl/src/main/java/org/apache/myfaces/config/annotation/DefaultAnnotationProvider.java

```diff
@@ -128,21 +128,23 @@ public DefaultAnnotationProvider()
     @Override
     public Map<Class<? extends Annotation>, Set<Class<?>>> getAnnotatedClasses(ExternalContext ctx)
     {
-        if (MyfacesConfig.getCurrentInstance(ctx).isUseCdiForAnnotationScanning())
+        //1. Use CDI
+        BeanManager beanManager = CDIUtils.getBeanManager(ctx);
+        if(beanManager != null)  // shouldnt be null, but needed for our mock junit tests
```

Comment:
@tandraschko  Is it okay if I add a null check here?  Otherwise, we'd have to handle the failing `mock` junit tests.
- skip them
- find some way to enable CDI ( looks tedious as we'll need to create lot of mock objects to replicate the CDIAnnotationDelegateInjectionProvider class)

Reply:
yep